tools/mpremote/mpremote/mip.py

```diff
@@ -102,16 +102,29 @@ def _download_file(transport, url, dest):
 
 
 def _install_json(transport, package_json_url, index, target, version, mpy):
-    try:
-        with urllib.request.urlopen(_rewrite_url(package_json_url, version)) as response:
-            package_json = json.load(response)
-    except urllib.error.HTTPError as e:
-        if e.status == 404:
-            raise CommandError(f"Package not found: {package_json_url}")
-        else:
-            raise CommandError(f"Error {e.status} requesting {package_json_url}")
-    except urllib.error.URLError as e:
-        raise CommandError(f"{e.reason} requesting {package_json_url}")
+    if (
+        package_json_url.startswith("http://")
+        or package_json_url.startswith("https://")
+        or package_json_url.startswith("github:")
```

Comment:
This probably needs to include gitlab as well.  And probably best to make a helper function so this code is not repeated with that in `_install_package`.

Maybe:
```py
allowed_prefix = ("http://", "https://", "github:", "gitlab:")
...
if package_json_url.startswith(allowed_prefix): ...
```

Reply:
Done. I have gone with your suggestion of

```
allowed_prefix = ("http://", "https://", "github:", "gitlab:")
...
if package_json_url.startswith(allowed_prefix):
...
if paclage.startswith(allowed_prefix):
```

rather than a helper function.